lib/clib/Corres_UL_C.thy

```diff
@@ -309,6 +309,14 @@ lemma ccorres_to_vcg:
   apply fastforce
   done
 
+lemma ccorres_to_vcg_Normal:
+  "\<lbrakk>ccorres_underlying srel \<Gamma> rrel xf arrel axf P P' [] a c; no_fail Q a\<rbrakk>
+   \<Longrightarrow> \<Gamma> \<turnstile> {s'. P s \<and> Q s \<and> s' \<in> P' \<and> (s, s') \<in> srel} c UNIV"
```

Comment:
I don't understand this one... is this equivalent to saying that it isn't UNIV? where does this fall out of?

Reply:
I have used this rule with the `ccorres` rules for `While` - when the C loop condition is just `UNIV`, the `ccorres_While` rule will ask you to show a `hoarep` goal with the postcondition just `UNIV`, which just means that the execution will result in a `Normal` state. This new rule allows us to do this fairly easily from a `ccorres` rule for the loop body.

Reply:
Fascinating. Does it really have to be a proper subset then?

Reply:
Sorry, what do you mean by a proper subset?

Reply:
Upon closer examination there seems to be some font rendering issue happening on my side that made the `c UNIV` look like `\<subset> UNIV`. Fortunately I can see the `c` now, and apologise for the noise.